src/System.Private.CoreLib/shared/System/SByte.cs

```diff
@@ -66,7 +66,7 @@ public bool Equals(sbyte obj)
         // Gets a hash code for this instance.
         public override int GetHashCode()
         {
-            return ((int)m_value ^ (int)m_value << 8);
+            return (int)m_value;
```

Comment:
nit: Would be nice to remove the unnecessary cast to `(int)` since C# will do it implicitly.

Reply:
I have pushed the fix for this.